lib/shared-method.js

```diff
@@ -440,16 +444,17 @@ function convertValueToTargetType(argName, value, targetType) {
  * @returns {String} The type name
  */
 
-SharedMethod.getType = function(val) {
+SharedMethod.getType = function(val, targetType) {
   var type = typeof val;
 
   switch (type) {
     case 'undefined':
     case 'boolean':
-    case 'number':
     case 'function':
     case 'string':
       return type;
+    case 'number':
+      return Number.isInteger(val) ? (targetType || 'integer') : 'number';
```

Comment:
Can we change/simplify this condition please? I find it difficult to understand correctly.

```js
return targetType === 'integer' && Number.isInteger(val) ? 'integer' : 'number';
```

Also please add a unit-test to describe how `number` value `2.0` is handled.